bin/apisix

```diff
@@ -52,7 +52,7 @@ if script_path:sub(1, 2) == './' then
         error("failed to fetch current path")
     end
 
-    if string.match(apisix_home, '^/[root][^/]+') then
+    if string.match(apisix_home .. "/", '^[/root]*/$') then
```

Comment:
I do not why the new regex can work correctly :(

Reply:
> @totemofwolf what is your current home path?
> 
> we can print variable `apisix_home`. I can not understand why you got the warning message.


```

➜  /root1 pwd
/root1


    -- if string.match(apisix_home .. "/", '^[/root]*/$') then
    if string.match(apisix_home, '^/[root][^/]+') then
            is_root_path = true
    end
...
end
print("apisix_home: ", apisix_home)

➜  /root1 make init
./bin/apisix init
apisix_home: 	/root1
Warning! Running apisix under /root is only suitable for development environments and it is dangerous to do so. It is recommended to run APISIX in a directory other than /root. 
./bin/apisix init_etcd
apisix_home: 	/root1

```

----


`if string.match(apisix_home .. "/", '^[/root]*/$') then` works both for `/root1`(no warning) and `/root` (with warning) which is the correct logic ;-)

```

➜  /root1 pwd
/root1

    if string.match(apisix_home .. "/", '^[/root]*/$') then
    -- if string.match(apisix_home, '^/[root][^/]+') then
            is_root_path = true
    end

...
end
print("apisix_home: ", apisix_home)

➜  /root1 make init
./bin/apisix init
apisix_home: 	/root1
./bin/apisix init_etcd
apisix_home: 	/root1

```

Reply:
Wow, your example is very special

Reply:
`if string.match(apisix_home .. "/", '^/root/') then` 

You can make a try in this way, this way is better.

Reply:
> `if string.match(apisix_home .. "/", '^/root/') then`
> 
> You can make a try in this way, this way is better.


```
    if string.match(apisix_home .. "/", '^[/root]*/$') then
    -- if string.match(apisix_home, '^/[root][^/]+') then
    if string.match(apisix_home .. "/", '^/root/') then
            is_root_path = true
    end
```

It works except ` if string.match(apisix_home, '^/[root][^/]+') then`

Reply:
I think only `'^/root/'` is right, it means the prefix should be `/root/`.

`^[/root]*/$` is wrong, here is an example:

```shell
$ resty -e 'ngx.say(string.match("/roo/", "^[/root]*/$"))'
/roo/
```